### PR TITLE
Better structure for some messages

### PR DESCRIPTION
**Pull request by MahdiBM**

### CODE REVIEW

File: Sources/Penny/Handlers/AuditLogHandler.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        context: HandlerContext
+    ) {
+        self.event = event
+        self.context = context
+        self.logger[metadataKey: "event"] = "\(event)"
+    }
+
+    func handle() async throws {
+        switch event.action {
+        case .memberBanAdd:
```
Comment (MahdiBM) [inline Sources/Penny/Handlers/AuditLogHandler.swift:27]: Report bans. I know it's in the audit-log too, but this is more convenient to have a message about in the moderators channel.

File: Sources/Penny/Handlers/MessageDeleteHandler.swift [hunk 2]
```diff
@@ -41,29 +44,53 @@ struct MessageDeleteHandler {
...
+                ]
+            )],
+            files: [.init(
+                data: ByteBuffer(data: jsonData),
+                filename: messageName
+            )],
+            attachments: [.init(
+                index: 0,
+                filename: messageName
             )]
```
Comment (MahdiBM) [inline Sources/Penny/Handlers/MessageDeleteHandler.swift:94]: include the whole message as an attachment for further investigations if needed.

File: Sources/Penny/Handlers/MessageDeleteHandler.swift [hunk 3]
```diff
@@ -41,29 +44,53 @@ struct MessageDeleteHandler {
...
-        let avatarURL = author.avatar.map {
-            CDNEndpoint.userAvatar(
-                userId: author.id,
-                avatar: $0
-            ).url
-        }
+        let member = messageCreate.member
+        let avatarURL = member?.uiAvatarURL ?? author.uiAvatarURL
+        let username = member?.uiName ?? author.username
+        let messageName = "message_\(messageCreate.id.rawValue)"
```
Comment (MahdiBM) [inline Sources/Penny/Handlers/MessageDeleteHandler.swift:55]: intentionally doesn't contain the .json extension so Discord doesn't expand the text on Desktop which looks bad.

File: Sources/Penny/Handlers/MessageDeleteHandler.swift [hunk 4]
```diff
@@ -41,29 +44,53 @@ struct MessageDeleteHandler {
...
+                        value: author.id.rawValue,
+                        inline: true
+                    ),
+                    .init(
+                        name: "Channel",
+                        value: DiscordUtils.mention(id: messageCreate.channel_id),
+                        inline: true
+                    )
+                ]
+            )],
```
Comment (MahdiBM) [inline Sources/Penny/Handlers/MessageDeleteHandler.swift:86]: change structure

**Review by gwynne — APPROVED** [review]

**Review by gwynne — APPROVED** [review]

**Pull request merged**